docs/ref/forms/api.txt

```diff
@@ -1464,26 +1473,39 @@ Methods of ``BoundField``
         >>> print(bound_form["subject"].value())
         hi
 
+.. _custom-boundfield:
+
 Customizing ``BoundField``
 ==========================
 
-If you need to access some additional information about a form field in a
-template and using a subclass of :class:`~django.forms.Field` isn't
-sufficient, consider also customizing :class:`~django.forms.BoundField`.
+.. attribute:: Form.bound_field_class
+
+.. versionadded:: 5.2
+
+There are multiple ways to use a custom :class:`.BoundField`:
+
+* A project can define a project-wide :class:`.BoundField` class using
+  :attr:`.BaseRenderer.bound_field_class` and a custom
+  :setting:`FORM_RENDERER`.
 
-A custom form field can override ``get_bound_field()``:
+* A form can override :attr:`.Form.bound_field_class` or set the
+  ``bound_field_class`` argument in the constructor.
 
-.. method:: Field.get_bound_field(form, field_name)
+* A custom form field can override :attr:`.Field.bound_field_class` or set the
+  ``bound_field_class`` argument in the constructor.
 
-    Takes an instance of :class:`~django.forms.Form` and the name of the field.
-    The return value will be used when accessing the field in a template. Most
-    likely it will be an instance of a subclass of
-    :class:`~django.forms.BoundField`.
+* Lastly, a custom form field can override :meth:`.Field.get_bound_field()`
+  though any of the previous options is preferred.
 
-If you have a ``GPSCoordinatesField``, for example, and want to be able to
+You may want to use a custom :class:`.BoundField` if you need to access some
+additional information about a form field in a template and using a subclass of
+:class:`~django.forms.Field` isn't sufficient.
```

Comment:
I was going to suggest that we're losing something by not having this at the top of the section of the docs. However, this is the reference documentation and I don't think that's reasonable. However, it does feel like some of the other new documentation may be better suited for the how to guide section. This is because you're showing what types of things are possible when the `BoundField` is customized.

I'm bringing this up more to see what others think rather than as a blocker to merge.

Reply:
I think it would be okay to merge as is and improve the docs later. Everything is covered, it's a matter of structure now.

Reply:
Sounds good to me!

Reply:
I agree with Adam. The current proposal aligns and flows well into the existing docs. If we want a bigger refactoring of this doc page, we can do so in a future PR (which would be only restricted by the string freeze on release candidate date).